shell/platform/android/test/io/flutter/embedding/android/FlutterActivityAndFragmentDelegateTest.java

```diff
@@ -343,7 +347,35 @@ public void itExecutesDartEntrypointProvidedByHost() {
     delegate.onStart();
 
     // Verify that the host's Dart entrypoint was used.
-    verify(mockFlutterEngine.getDartExecutor(), times(1)).executeDartEntrypoint(eq(dartEntrypoint));
+    verify(mockFlutterEngine.getDartExecutor(), times(1))
+        .executeDartEntrypoint(eq(dartEntrypoint), any(List.class));
+  }
+
+  @Test
+  public void itExecutesDartEntrypointWithArgsProvidedByHost() {
+    // ---- Test setup ----
+    // Set Dart entrypoint parameters on fake host.
+    when(mockHost.getAppBundlePath()).thenReturn("/my/bundle/path");
+    when(mockHost.getDartEntrypointFunctionName()).thenReturn("myEntrypoint");
+    List<String> dartEntrypointArgs = new ArrayList<String>(Arrays.asList("foo", "bar"));
+    when(mockHost.getDartEntrypointArgs()).thenReturn(dartEntrypointArgs);
+
+    // Create the DartEntrypoint that we expect to be executed
+    DartExecutor.DartEntrypoint dartEntrypoint =
+        new DartExecutor.DartEntrypoint("/my/bundle/path", "myEntrypoint");
+
+    // Create the real object that we're testing.
+    FlutterActivityAndFragmentDelegate delegate = new FlutterActivityAndFragmentDelegate(mockHost);
+
+    // --- Execute the behavior under test ---
+    // Dart is executed in onStart().
+    delegate.onAttach(RuntimeEnvironment.application);
+    delegate.onCreateView(null, null, null, 0, true);
+    delegate.onStart();
+
+    // Verify that the host's Dart entrypoint was used.
+    verify(mockFlutterEngine.getDartExecutor(), times(1))
+        .executeDartEntrypoint(eq(dartEntrypoint), eq(dartEntrypointArgs));
```

Comment:
this could use `any(DartExecutor.DartEntrypoint.class)`, since this test is for the entrypoint args only

Reply:
done